src/dumbo/src/tcp/endpoint.rs

```diff
@@ -299,16 +303,36 @@ impl Endpoint {
     }
 }
 
+/// `parse_request` parses the request bytes and builds a `micro_http::Response` by the given callback function.
+pub fn parse_request(
+    request_bytes: &[u8],
+    callback: fn(Result<Request, RequestError>) -> Response,
+) -> Response {
```

Comment:
This `parse_request` needs to map a stream of bytes to a `micro_http::Request`. If any error is raised during this mapping, this error should be sent to upper levels. 

```suggestion
) -> Result<Response, RequestError> {
```